### PR TITLE
Fixed ghost snippet behavior

### PR DESCRIPTION
**Pull request by danrot**
__tasks:__

- [ ] test coverage
- [ ] gather feedback for my changes

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | none
| BC breaks        | none
| Documentation PR | none

### CODE REVIEW

File: src/Sulu/Bundle/SnippetBundle/Tests/Functional/Controller/SnippetControllerTest.php [hunk 1]
```diff
@@ -267,7 +272,7 @@ public function testDeleteReferenced()
     private function loadFixtures()
     {
         // HOTELS
-        $req = ContentMapperRequest::create()
+        $request = ContentMapperRequest::create()
```
Comment (dantleech) [inline src/Sulu/Bundle/SnippetBundle/Tests/Functional/Controller/SnippetControllerTest.php]: We can now start pro-actively replacing things like this with the document manager:

````
$doc = $documentManager->create('snippet');
$doc->setStructureType('hotel');
$doc->getTitle('Le frane..');
$documentManager->persist($doc, 'de');
$documentManager->flush();
````

But might be better to make that policy after the merge.

**Comment by dantleech** [comment]
Looks ok at a glance, lets wait for the tests.

File: src/Sulu/Bundle/SnippetBundle/Tests/Functional/Controller/SnippetControllerTest.php (hunk 1 above)
Comment (danrot) [inline src/Sulu/Bundle/SnippetBundle/Tests/Functional/Controller/SnippetControllerTest.php]: I'll do it, why should we wait with that :)

File: src/Sulu/Bundle/SnippetBundle/Tests/Functional/Controller/SnippetControllerTest.php [hunk 2]
```diff
@@ -267,55 +262,37 @@ public function testDeleteReferenced()
...
-            ->setData(array(
-                'title' => 'Ford',
-            ));
-        $this->contentMapper->saveRequest($req);
+        $car1 = $this->documentManager->create('snippet');
+        $car1->setStructureType('car');
+        $car1->setTitle('Skoda');
+        $this->documentManager->persist($car1, 'de');
+
+        $car1 = $this->documentManager->create('snippet');
```
Comment (dantleech) [inline src/Sulu/Bundle/SnippetBundle/Tests/Functional/Controller/SnippetControllerTest.php]: technically this is `car2`.. I would just call them all `car`

File: src/Sulu/Component/Content/Document/Subscriber/FallbackLocalizationSubscriber.php [hunk 3]
```diff
@@ -151,16 +147,16 @@ public function getAvailableLocalization(NodeInterface $node, $document, $locale
...
-     * @param NodeInterface $node
-     * @param mixed $locale
+     * @param string $webspaceName
+     * @param string[] $availableLocales
+     * @param string $locale
+     *
+     * @return string
      */
-    private function getWebspaceLocale($document, NodeInterface $node, $locale)
+    private function getWebspaceLocale($webspaceName, $availableLocales, $locale)
```
Comment (dantleech) [inline src/Sulu/Component/Content/Document/Subscriber/FallbackLocalizationSubscriber.php]: `$webspaceName, array $availableLocales`

**Comment by dantleech** [comment]
Oops. I merged it accidentally. Fix the above minor comments if you want.

**Comment by danrot** [comment]
fixed in 1652fe0

**Pull request merged**